Feature/friendship functionalities

WretchedBunny opened the pull request.

scherepiuk committed:
GOEM-120: Notification screen scaffold
Two new models: FriendshipRequest and Notification
Notifications' repository and data sources

scherepiuk committed:
GEOM-120: Notification screen completed
Organized UI components into corresponding packages

WretchedBunny committed:
conflict resoled

WretchedBunny committed:
Implementation of Friendship functionalities
Created classes and interface for database operations (and method for retrieval of friendshipRequest document).
Created Friendship model
Created method for creation of the FriendshipState based on friendshipRequest document.(further development is required)

scherepiuk committed:
GEOM-120: Rewrote repository and data sources methods to use Flow

scherepiuk committed:
Fetching images for search results

scherepiuk committed:
Extracted "signUp" and "signIn" implementations into an "auth" method

WretchedBunny committed:
feature(other-profile): Continuation of implementation of Friendship functionalities
1. Added new methods to FriendshipDataSource.kt such user related activities: add, remove, updateStatus, updateNotifications
2. Added corresponding implementation of the methods to FriendshipRemoteDataSource.kt
3. Added methods calls to FriendshipRepository.kt.
4. Friendship.kt was renamed to FriendshipRequest.kt in order to match corresponding collection in Fb. enum class FriendshipStatus was simplified.
5. FriendshipState.kt was refactored. Objects of sealed class were changed for classes for sake of passing FriendshipRepository.kt and accessing methods of mentioned class instance. All state's methods were implemented. REQUIRED: Analysis of the current StateMachine implementation.
6. Added userId parameter for composable ProfileButtonsRow.kt, launch coroutine builders and required methods calls.
7. ProfileScreen.kt changed ProfileButtonsRow.kt call.
8. ProfileUiState.kt added friendshipRequest property
9. ProfileViewModel.kt refactor of createFriendshipState() method.

WretchedBunny committed:
feature(other-profile): Continuation of implementation of Friendship functionalities
1. Added new methods to FriendshipDataSource.kt such user related activities: add, remove, updateStatus, updateNotifications
2. Added corresponding implementation of the methods to FriendshipRemoteDataSource.kt
3. Added methods calls to FriendshipRepository.kt.
4. Friendship.kt was renamed to FriendshipRequest.kt in order to match corresponding collection in Fb. enum class FriendshipStatus was simplified.
5. FriendshipState.kt was refactored. Objects of sealed class were changed for classes for sake of passing FriendshipRepository.kt and accessing methods of mentioned class instance. All state's methods were implemented. REQUIRED: Analysis of the current StateMachine implementation.
6. Added userId parameter for composable ProfileButtonsRow.kt, launch coroutine builders and required methods calls.
7. ProfileScreen.kt changed ProfileButtonsRow.kt call.
8. ProfileUiState.kt added friendshipRequest property
9. ProfileViewModel.kt refactor of createFriendshipState() method.

scherepiuk committed:
Merge pull request #19 from SergeyCherepiuk/feature/searchProfilePicturesFix

Fetching images for search results

scherepiuk committed:
Merge pull request #20 from SergeyCherepiuk/feature/googleAuthMethodsRefactor

GoogleAuth methods refactor

scherepiuk commented on app/src/main/java/com/example/geomate/ui/screens/profile/ProfileScreen.kt:
```diff
@@ -115,6 +114,8 @@ fun ProfileScreen(
         viewModel.fetchProfilePicture(userId)
         viewModel.fetchUser(userId)
+        viewModel.fetchFriendship(userId)
+
```
Remove white-space

scherepiuk commented on app/src/main/java/com/example/geomate/ui/screens/profile/ProfileUiState.kt:
```diff
@@ -1,11 +1,13 @@
 data class ProfileUiState(
     val profilePictureUri: Uri? = null,
     val user: User = User(),
+    val friendshipRequest: FriendshipRequest? = FriendshipRequest(),
```
`= null`

WretchedBunny committed:
fix: Minor fixes for Friendship functionalities.
1. FriendshipRemoteDataSource.kt renamed to FriendshipRemoteRequestDataSource.kt. Renamed according interface. Implemented new method to get FriendshipRequest.kt.
2.Removed None status from FriendshipStatus due to lack of usage for it.
3. General clean-up and code reduction.

WretchedBunny committed:
fix: Minor fixes for Friendship functionalities.
1. FriendshipRemoteDataSource.kt renamed to FriendshipRemoteRequestDataSource.kt. Renamed according interface. Implemented new method to get FriendshipRequest.kt.
2.Removed None status from FriendshipStatus due to lack of usage for it.
3. General clean-up and code reduction.

scherepiuk commented on app/src/main/java/com/example/geomate/data/datasources/FriendshipRemoteRequestDataSource.kt:
```diff
@@ -11,7 +11,8 @@ import kotlinx.coroutines.flow.Flow
 import kotlinx.coroutines.tasks.await
 
-class FriendshipRemoteDataSource(private val fireStore: FirebaseFirestore) : FriendshipDataSource {
+class FriendshipRemoteRequestDataSource(private val fireStore: FirebaseFirestore) :
```
`FriendshipRequestRemoteDataSource`

WretchedBunny committed:
fix: Minor fixes for Friendship functionalities.
1. ProfileUiState friendshipRequest was assigned to null
2. FriendshipRemoteRequestDataSource was renamed to FriendshipRequestRemoteDataSource

WretchedBunny committed:
fix: Minor fixes for Friendship functionalities.
1. ProfileUiState friendshipRequest was assigned to null
2. FriendshipRemoteRequestDataSource was renamed to FriendshipRequestRemoteDataSource

WretchedBunny committed:
Merge pull request #22 from SergeyCherepiuk/feature/notificationScreen

Notification screen

WretchedBunny committed:
resolved conflicts

WretchedBunny committed:
Merge remote-tracking branch 'origin/feature/friendshipFunctionalities' into feature/friendshipFunctionalities

# Conflicts:
#	app/src/main/java/com/example/geomate/ui/navigation/NavGraph.kt
#	app/src/main/java/com/example/geomate/ui/screens/profile/ProfileScreen.kt
#	app/src/main/java/com/example/geomate/ui/screens/profile/components/Buttons.kt
